sdk/core/Azure.Core/src/HttpMessage.cs

```diff
@@ -14,7 +14,7 @@ namespace Azure.Core
     /// </summary>
     public sealed class HttpMessage : IDisposable
     {
-        private Dictionary<string, object>? _properties;
+        private Dictionary<Type, object>? _typeProperties;
```

Comment:
Are we going to move all SDK usages of the message property APIs from strings to Types?

Reply:
I think we should in the instances where it makes sense.

Reply:
It might be worth to proactively move at least a couple of usages to make sure that the new approach works as intended/envisioned.